src/moonlink/src/table_handler.rs

```diff
@@ -526,6 +527,23 @@ impl TableHandler {
                                 }
                             }
                         }
+                        TableEvent::FlushResult { xact_id, flush_result } => {
+                            match flush_result {
+                                Some(Ok(disk_slice)) => {
+                                    if let Some(xact_id) = xact_id {
+                                        table.apply_stream_flush_result(xact_id, disk_slice);
+                                    } else {
+                                        table.apply_flush_result(disk_slice);
+                                    }
+                                }
+                                Some(Err(e)) => {
```

Comment:
let's unwrap it to keep the same behavior (prev sync flush unwraps directly)

My concern is:
- for now, for failed async flush, when an error happens we do nothing; no snapshot/mooncake table state update, no crash
- for chaos test it failed right away in the past, since event handler thread crashes; now it needs to wait for timeout on CI, and with no events printed out